core/src/main/java/org/ldaptive/provider/jndi/JndiConnectionFactory.java

```diff
@@ -51,7 +51,9 @@ public JndiConnectionFactory(
     environment = Collections.unmodifiableMap(env);
     classLoader = cl;
     if (ThreadLocalTLSSocketFactory.class.getName().equals(environment.get(JndiProvider.SOCKET_FACTORY))) {
-      threadLocalSslConfig = new ThreadLocalTLSSocketFactory().getSslConfig();
+      final ThreadLocalTLSSocketFactory sf = new ThreadLocalTLSSocketFactory();
+      threadLocalSslConfig = sf.getSslConfig();
+      sf.removeSslConfig();
```

Comment:
The appearance here in the constructor raised a red flag for me, but after having studied the surrounding code I believe I understand that it could be considered correct. The consequence of clearing the thread local here implies there are no other consumers of it, which I assume is the nominal case, but it does beg the question of how it gets set in the first place. I would think it would be safer to use the finalizer pattern to clean up these kinds of resources; is that an option?

Reply:
See https://github.com/vt-middleware/ldaptive/blob/master/core/src/main/java/org/ldaptive/provider/jndi/JndiProvider.java#L176 for a gnarly example of how this works.

Reply:
So I suppose the thread local could be removed in a finally block in `JndiProvider`, rather than in the `JndiConnectionFactory` constructor internals.

Reply:
The component that sets up thread locals should be responsible for cleaning them up, and your suggestion to put the cleanup code in `JndiProvider` is consistent with that advice.

Reply:
Pushed a fix for this.